File: missing-number/missing-number.java
class Solution {
    public int missingNumber(int[] nums) {
        Set<Integer> mySet = new HashSet<>();
        
        for (int i = 0; i < nums.length; i ++){
            if (!mySet.contains(nums[i])){
                mySet.add(nums[i]);
            }
        }
        
        int expectedNum = nums.length + 1;
        for (int i = 0; i < expectedNum; i ++){
            if (!mySet.contains(i)){
                return i;
            }
        }
        return -1;
    
        
    }
}

File: missing-number/missing-number.py
class Solution:
    def missingNumber(self, nums: List[int]) -> int:
        seen = set()
        expected = len(nums) + 1
        ans = 0
        
        for i in nums:
            if i not  in seen:
                seen.add(i)
                
        print(seen)
                
        for i in range(expected):
            if i not in seen:
                ans = i 
                
        return ans
                
            
                
    
        